src/test/java/TestAhoCorasickDoubleArrayTrie.java

```diff
@@ -101,6 +102,51 @@ public void hit(int begin, int end, String value)
         });
     }
 
+    private static class CountHits implements AhoCorasickDoubleArrayTrie.IHitCancellable<String> {
+        private int count = 0;
+        private boolean countAll = true;
+
+        public CountHits(boolean countAll) {
+            this.countAll = countAll;
+        }
+
+        public int getCount() {
+            return count;
+        }
+
+        @Override
+        public boolean hit(int begin, int end, String value) {
+            count += 1;
+            return countAll;
+        }
+    }
+
+    public void testCancellation() throws Exception {
+        // Collect test data set
+        TreeMap<String, String> map = new TreeMap<String, String>();
+        String[] keyArray = new String[]
+                {
+                        "foo",
+                        "bar"
+                };
+        for (String key : keyArray)
+        {
+            map.put(key, key);
```

Comment:
You could create the map using `streams` instead, which is nicer than creating it empty and then having a for lop mutating it.